src/session.ts

```diff
@@ -40,6 +40,18 @@ async function updateSession(request: NextRequest, debug: boolean, middlewareAut
     return pathRegex.exec(request.nextUrl.pathname);
   });
 
+  const url = new URL(WORKOS_REDIRECT_URI);
+
+  if (
+    middlewareAuth.enabled &&
+    url.pathname === request.nextUrl.pathname &&
+    !middlewareAuth.unauthenticatedPaths.includes(url.pathname)
+  ) {
+    throw new Error(
```

Comment:
instead of throwing an error, i think we should just push the REDIRECT_URL onto unauthenticatedPaths (could probably be done at startup instead of in the middleware).